templates/problem/manage_submission.html

```diff
@@ -0,0 +1,153 @@
+                        return;
+                    } else if (start > end) {
+                        alert("{{ _('End ID must be after start ID.') }}");
+                        return;
+                    }
+                }
+                $(this).parents('form').submit();
+            });
+
+            $use_id.change(function () {
+                $('#by-range-filter').find('input').prop('disabled', !this.checked);
+            });
+        });
+    </script>
+{% endblock %}
+
+{% block body %}
+    {% include "messages.html" %}
+
+    <div class="panes">
+        <div class="pane">
+            <h3>{{ _('Rejudge Submissions') }}</h3>
+            <form action="{{ url('problem_submissions_rejudge', problem.code) }}" method="post">
+                {% csrf_token %}
+                <div class="control-group">
+                    <label><input id="by-range-check" type="checkbox" name="use_range" value="on">
+                        {{ _('Filter by ID:') }}</label>
+                    <table id="by-range-filter" class="table">
+                        <tr>
+                            <th><label for="by-range-start">{{ _('Starting ID:') }}</label></th>
+                            <td><input id="by-range-start" name="start" type="number" disabled></td>
+                        </tr>
+                        <tr>
+                            <th><label for="by-range-end">{{ _('Ending ID:') }}</label></th>
+                            <td><input id="by-range-end" name="end" type="number" disabled></td>
+                        </tr>
+                    </table>
+                    <p>{{ _('This range includes both endpoints.') }}</p>
+                </div>
+                <div class="control-group">
+                    <label for="by-lang-filter">{{ _('Filter by language:') }}</label>
+                    <select id="by-lang-filter" name="language" multiple>
+                        {% for id, name in languages %}
+                            <option value="{{ id }}">{{ name }}</option>
+                        {% endfor %}
+                    </select>
+                </div>
+                <div class="control-group">
+                    <label for="by-result-filter">{{ _('Filter by result:') }}</label>
+                    <select id="by-result-filter" name="result" multiple>
+                        {% for name in results %}
+                            <option>{{ name }}</option>
+                        {% endfor %}
+                    </select>
+                </div>
+                <a id="rejudge-selected" class="unselectable button full" href="#">
+                    {{ _('Rejudge selected submissions') }}
+                </a>
+            </form>
+        </div>
+
+        <div class="pane">
+            <h3>{{ _('Rescore Everything') }}</h3>
+            <p id="rescore-warning">{{ _('This will rescore %(count)d submissions.', count=submission_count) }}</p>
```

Comment:
Should we also add a warning for rejudges?

Reply:
This is done.